coop_badge_reader/models/res_partner.py

```diff
@@ -64,83 +65,84 @@ def compute_next_shift_time(self):
-                'bootstrap_cooperative_state':
-                partner.bootstrap_cooperative_state,
+                "partner_id": partner.id,
+                "cooperative_state": partner.cooperative_state,
+                "action": action,
+                "bootstrap_cooperative_state":
+                    partner.bootstrap_cooperative_state,
             })
-            if partner_alert_mail_template and action == 'in':
-                partner_alerts = partner_alert_obj.search([
-                    ('expected_member_id', '=', partner.id),
-                    ('state', '=', 'open'),
-                ])
+            if partner_alert_mail_template and action == "in":
+                partner_alerts = partner_alert_obj.search(
+                    [
+                        ("expected_member_id", "=", partner.id),
+                        ("state", "=", "open"),
+                    ]
+                )
 
                 for partner_alert in partner_alerts:
                     partner_alert_mail_template.send_mail(
-                        res_id=partner_alert.id, force_send=True)
+                        res_id=partner_alert.id, force_send=True
+                    )
 
     @api.multi
     def action_grace_partner(self):
-        '''
+        """
         @Function call when gracing a partner:
             Creating Extension for partner
                 - Start Date: Current Date
                 - End Date:
                     Current Date + Duration
                     If End Date exceeds the next Shift Time, End Date will
                     be the next shift time
-        '''
+        """
         self.ensure_one()
-
-        current_extension = self.extension_ids.filtered('current_extension')
-
+        current_extension = self.extension_ids.filtered("current_extension")
         if current_extension:
             return current_extension.date_stop
-
         # Only grace extensions for suspended user with no extension
-        if self.cooperative_state != 'suspended':
+        if self.cooperative_state != "suspended":
             return False
-
-        shift_ext_env = self.env['shift.extension']
-        ext_type_env = self.env['shift.extension.type']
+        shift_ext_env = self.env["shift.extension"]
+        ext_type_env = self.env["shift.extension.type"]
         date_start_str = fields.Date.context_today(self)
-        grace_ext_type = ext_type_env.sudo().search([
-            ('is_grace_period', '=', True)], limit=1)
-
+        grace_ext_type = ext_type_env.sudo().search(
+            [("is_grace_period", "=", True)], limit=1
+        )
         if not grace_ext_type:
             return False
-
         # Create new extension
-        date_stop = datetime.strptime(date_start_str, DF) + \
-            timedelta(days=grace_ext_type.duration)
+        date_stop = datetime.strptime(date_start_str, DF) + timedelta(
+            days=grace_ext_type.duration
+        )
         date_stop_str = date_stop.strftime(DF)
```

Comment:
In `12.0` date and datetime fields can be used as date and datetime python objects.
Can we improve the code to avoid transforming the date to strings, and maybe remote `date_start_str` and `date_stop_str`?

Reply:
There are still many strftime / strptime. I think it needs a bigger refactor to have it working with the new date/datetime fields. I mean, it's probably working now but it'll be good to avoid converting between date/str. What do you think about it?